### ISSUE TITLE
[BUG]: missing multiplication in `_modifiedHuberLoss` implementation in `ml/incr/binary-classification`

### ISSUE THREAD

**Issue by nakul-krishnakumar** [stdlib-js/stdlib#13111]
### Description

I was going through the wikipedia pages as well as other library implementations of loss functions to implement standalone loss functions and their gradients under `ml/base/loss/float64/*`. 
I noticed a possible bug in `_modifiedHuberLoss` function implemented under `ml/incr/binary-classification`

Current implementation:
```javascript
setReadOnly( Model.prototype, '_modifiedHuberLoss', function modifiedHuber( x, y ) {
	var eta;
	var d;

	eta = this[ this._learningRateMethod ]();
	this._regularize( eta );

	d = y * this._dot( x.data, x.strides[ 0 ], x.offset );
	if ( d < -1.0 ) {
		this._add( x, 4.0*eta*y );
	} else {
		this._add( x, eta*( y-(d*y) ) );
	}
	return this;
});
```
[Link](https://github.com/stdlib-js/stdlib/blob/8bf91614301abf9249ff3ea9379baa4bc6c450e0/lib/node_modules/%40stdlib/ml/incr/binary-classification/lib/model.js#L334-L348)

I believe in the `else` branch, it should be `this._add( x, eta*2*( y-(d*y) ) )` instead of `this._add( x, eta*( y-(d*y) ) )`. That is, it seems like it is missing a **multiplication with 2** here.

References:
[Wikipedia](https://en.wikipedia.org/wiki/Huber_loss#Variant_for_classification)
[sklearn](https://github.com/scikit-learn/scikit-learn/blob/fe2edb3cdbd75ae4e662fda67dcb19277258792b/sklearn/linear_model/_sgd_fast.pyx.tp#L145)

cc: @Planeshifter

### Questions

No.

### Reproduction

- a
- b
- c

### Expected Results

```math
\frac{\partial \ell}{\partial p} =
\begin{cases}
-4y & \text{if } yp < -1 \\
-2y(1 - yp) & \text{if } -1 \le yp \le 1 \\
0 & \text{if } yp > 1
\end{cases}
```

### Actual Results

```shell

```

### Environments

N/A

### Checklist

- [x] Read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.

**Comment by A-S-Manoj** [comment]
Confirmed — checked against sklearn's _sgd_fast.pyx, which has 2.0 * (1.0 - z) * -y for the quadratic branch. That 2 comes from differentiating (1-yf)², so the update should be 2η(y−dy), matching the issue.
I'd be happy to open a PR for this.

**Comment by kgryte** [comment]
@Planeshifter Perhaps you had a rationale for omitting the `2` factor?

**Comment by nakul-krishnakumar** [comment]
I can see a similar convention of omitting the `2` factor for `squared-hinge-loss` too in `ml/incr/binary-classification`. [Ref](https://github.com/stdlib-js/stdlib/blob/f0362db28573c53db1b750ffb1898406f34c061b/lib/node_modules/%40stdlib/ml/incr/binary-classification/lib/model.js#L482-L503)

**Comment by Planeshifter** [comment]
@nakul-krishnakumar I believe you are correct and that the `2` factor is indeed missing.

Looking at the gradient equation, it seems we are also missing a `yp ≥ 1` clamp in the modified Huber loss implementation.  `_squaredHingeLoss` does clamp correctly if d < 1.0.

For reference, here is `scikit-learn`:

```py
    cdef double cy_gradient(self, double y, double p) noexcept nogil:
        cdef double z = p * y
        if z >= 1.0:
            return 0.0
        elif z >= -1.0:
            return 2.0 * (1.0 - z) * -y
        else:
            return -4.0 * y
```
(https://github.com/scikit-learn/scikit-learn/blob/fe2edb3cdbd75ae4e662fda67dcb19277258792b/sklearn/linear_model/_sgd_fast.pyx.tp#L140-L149)

Please feel free to open a fix PR.

**Comment by nakul-krishnakumar** [comment]
Thank you @Planeshifter, I will open a fixing PR shortly.
